conda_forge_tick/git_utils.py

```diff
@@ -0,0 +1,363 @@
+        Object for communicating with GitHub, if None build from $USERNAME
+        and $PASSWORD, defaults to None
+
+    Returns
+    -------
+    recipe_dir : str
+        The recipe directory
+    """
+    gh = ctx.gh
+    # first, let's grab the feedstock locally
+    upstream = feedstock_url(fctx=fctx, feedstock=feedstock, protocol=protocol)
+    origin = fork_url(upstream, ctx.github_username)
+    feedstock_reponame = feedstock_repo(fctx=fctx, feedstock=feedstock)
+
+    if pull_request or fork:
+        repo = gh.repository("conda-forge", feedstock_reponame)
+        if repo is None:
+            fctx.attrs["bad"] = "{}: does not match feedstock name\n".format(
+                fctx.package_name,
+            )
+            return False
+
+    # Check if fork exists
+    if fork:
+        try:
+            fork_repo = gh.repository(ctx.github_username, feedstock_reponame)
+        except github3.GitHubError:
+            fork_repo = None
+        if fork_repo is None or (hasattr(fork_repo, "is_null") and fork_repo.is_null()):
+            print("Fork doesn't exist creating feedstock fork...")
+            repo.create_fork()
+            # Sleep to make sure the fork is created before we go after it
+            time.sleep(5)
+
+    feedstock_dir = os.path.join(ctx.rever_dir, fctx.package_name + "-feedstock")
+    from conda_forge_tick.git_xonsh_utils import fetch_repo
+
+    if fetch_repo(
+        feedstock_dir=feedstock_dir, origin=origin, upstream=upstream, branch=branch,
+    ):
+        return feedstock_dir, repo
+    else:
+        return None
+
+
+def delete_branch(ctx: GithubContext, pr_json: LazyJson, dry_run: bool = False) -> None:
+    ref = pr_json["head"]["ref"]
+    if dry_run:
+        print("dry run: deleting ref %s" % ref)
+        return
+    name = pr_json["base"]["repo"]["name"]
+    token = ctx.github_password
+    deploy_repo = ctx.github_username + "/" + name
+    doctr_run(
+        [
+            "git",
+            "push",
+            "https://{token}@github.com/{deploy_repo}.git".format(
+                token=token, deploy_repo=deploy_repo,
+            ),
+            "--delete",
+            ref,
+        ],
+        token=token.encode("utf-8"),
+    )
+    # Replace ref so we know not to try again
+    pr_json["head"]["ref"] = "this_is_not_a_branch"
+
+
+@backoff.on_exception(
+    backoff.expo, (RequestException, Timeout), max_time=MAX_GITHUB_TIMEOUT,
+)
+def refresh_pr(
+    ctx: GithubContext,
+    pr_json: LazyJson,
+    gh: Optional[github3.GitHub] = None,
+    dry_run: bool = False,
+) -> Optional[dict]:
+    gh = ensure_gh(ctx, gh)
+    if not pr_json["state"] == "closed":
+        if dry_run:
+            print("dry run: refresh pr %s" % dict(pr_json))
+        else:
+            pr_obj = github3.pulls.PullRequest(dict(pr_json), gh)
+            pr_obj.refresh(True)
+            pr_obj_d = pr_obj.as_dict()
+            # if state passed from opened to merged or if it
```

Comment:
fix this statement, it isn't true. We don't delete after a day closed. The bot doesn't check closed PRs.

Reply:
Given that this is the phase where we do the refresh, this is where we determine if a pr that was previously active is now closed.  We *should* probably wait a bit here before closing the pr, and presumably did so in the past, but due to it returning the updated state, we likely never did the deletes and this idea got lost.

Reply:
I don't think this closes the PR, it just checks if the branch is ready for deletion, with ready being if the PR was merged. I'm not certain if it is worthwhile to delete branches that were not merged, since humans may have added code to that or want to reopen that PR. (Even though the bot will no longer respond to the re-opened PR, it would otherwise be fine)